components/dashboard/src/service/json-rpc-workspace-client.ts

```diff
@@ -160,4 +169,66 @@ export class JsonRpcWorkspaceClient implements PromiseClient<typeof WorkspaceSer
         result.workspace = workspace.workspace;
         return result;
     }
+
+    async getWorkspaceDefaultImage(
+        request: PartialMessage<GetWorkspaceDefaultImageRequest>,
+        _options?: CallOptions | undefined,
+    ): Promise<GetWorkspaceDefaultImageResponse> {
+        if (!request.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        const response = await getGitpodService().server.getDefaultWorkspaceImage({
+            workspaceId: request.workspaceId,
+        });
+        const result = new GetWorkspaceDefaultImageResponse();
+        result.defaultWorkspaceImage = response.image;
+        return result;
+    }
+
+    async sendHeartBeat(
+        request: PartialMessage<SendHeartBeatRequest>,
+        _options?: CallOptions | undefined,
+    ): Promise<SendHeartBeatResponse> {
+        if (!request.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        const workspace = await this.getWorkspace({ workspaceId: request.workspaceId });
```

Comment:
the same sendHeartBeat shoudl deal with it, i don't think we need spread logic

Reply:
This is because we need instanceId to call server sendHeartbeat